src/NodaTime/LocalDate.cs

```diff
@@ -760,6 +760,22 @@ public void Deconstruct(out int year, out int month, out int day)
             day = Day;
         }
 
+        /// <summary>
+        /// Deconstructs the current instance into its components.
+        /// </summary>
+        /// <param name="year">The year component of the date.</param>
+        /// <param name="month">The month component of the date.</param>
+        /// <param name="day">The day component of the date.</param>
+        /// <param name="calendar">The <see cref="CalendarSystem"/> associated with the date.</param>
+        [Pure]
+        public void Deconstruct(out int year, out int month, out int day, [NotNull]out CalendarSystem calendar)
```

Comment:
I think it's probably worth removing the `[NotNull]` attribute for the moment - aside from anything else, at the moment the documentation generator would add an `ArgumentNullException` XML doc. Will need to modify that - we haven't had much in the way of out parameters before...

(It's fine to use "amend" and force-push for that - no need for another commit.)